docs/contributing/development.md

```diff
@@ -0,0 +1,70 @@
+# Development documentation
+
+This document describes how to set up the development environment and how to develop Vald.
+
+## Set up environment
+
+### Prerequisites
+
+#### OS
+
+- When using Docker related environment, you can use any OS that supports Docker.
+- When using native environment, `Linux` is required.
+
+#### Architecture
+
+`amd64` is recommended because `NGT`, the vector search library we use, is optimized for `amd64`.
+But you can also build and test `Vald` on `arm64` with the same way as described below.
+
+### Devcontainer
+
+This is the easiest way to start developing `Vald`. You can just open our [devcontainer.json](https://github.com/vdaas/vald/blob/main/.devcontainer/devcontainer.json) with `VS Code` and go.
+
+### Other
+
+We don't officially have a setup documentation for now, but you can take a look at the [`Dockerfile`](https://github.com/vdaas/vald/blob/main/dockers/dev/Dockerfile).
+That's everything you need to build and test `Vald`, so you can use it as a reference.
+
+> If you would like to use the `Dockerfile` directlly, please note that `docker-in-docker` environment is required to run our E2E tests.
+> In devcontainer, [`VS Code` handles it for us](https://github.com/devcontainers/features/tree/main/src/docker-in-docker).
```

Comment:
To display it as a card view on the official website, you will need to alter the format.
It is a just suggestion, so you do not have to change.

Ref: https://vald.vdaas.org/docs/user-guides/upgrade-cluster/

Reply:
Let's check how it will look like on the web.